Standalone_GCBM/input_database/add_species_vol_to_bio.py

```diff
@@ -10,53 +11,68 @@
         # Spatial unit = 36 (British Colombia and Pacific maritime
-        conn.execute(
-            """
-            INSERT INTO vol_to_bio_factor_association (spatial_unit_id, species_id, vol_to_bio_factor_id,root_parameter_id)
-            VALUES (36, 8001, 801,1)
-            """)
+
+        with open('vol_to_bio_factor_association.csv','r') as fin3: # `with` statement available in 2.5+
+        # csv.DictReader uses first line in file for column headings by default
+        dr3 = csv.DictReader(fin3) # comma is default delimiter
+        to_db3 = [(i['spatial_unit_id'], i['species_id'], i['vol_to_bio_factor_id'], i['root_parameter_id']) for i in dr3]
+
+        cursor.executemany(
+            "INSERT INTO vol_to_bio_factor_association (spatial_unit_id, species_id, vol_to_bio_factor_id,root_parameter_id)
+            VALUES (?, ?, ?, ?)", to_db3
+            )
         # Create a new set of vol to bio parameters
         # See Bowdewyn et al to knwo how to set the parameters of the vol to bio equations
         # we have 70% (0.75) stemwood and a density of 0.47, so our a factor is 0.329
         # non merchantable and sampling factors set to zero
         # 
         logging.info("Adding Vol to Bio parameters")
-        conn.execute(
-            """
-            INSERT INTO vol_to_bio_factor
-            SELECT 801 AS id, 0.329 AS a, 1 AS b,
-            0 AS a_nonmerch, 1 AS b_nonmerch, 1 AS k_nonmerch, 1 AS cap_nonmerch,
-            0 AS a_sap, 1 AS b_sap, 1 AS k_sap, 1 AS cap_sap,
-            -1.945910149 AS a1, 0 AS a2, 0 AS a3,
-            -1.540445041 AS b1, 0 AS b2, 0 AS b3,
-            -2.63905733 AS c1, 0 AS c2, 0 AS c3,
-            0.1 AS min_volume, 1000000 AS max_volume,
-            0.50249 AS low_stemwood_pop, 0.70391 AS high_stemwood_pop,
-            0.09492 AS low_stembark_prop, 0.11402 AS high_stembark_prop,
-            0.26098 AS low_branches_prop, 0.14622 AS high_branches_pop,
-            0.1416 AS low_foliage_prop, 0.03585 AS high_foliage_prop
-            """)
-        
\ No newline at end of file
+        with open('species.csv','r') as fin4: # `with` statement available in 2.5+
+        # csv.DictReader uses first line in file for column headings by default
+        dr4 = csv.reader(fin4) # comma is default delimiter
+
+        cursor.executemany(
+            "INSERT INTO vol_to_bio_factor
+            SELECT 801 AS ?, 0.329 AS ?, 1 AS ?,
+            0 AS ?, 1 AS ?, 1 AS ?, 1 AS ?,
+            0 AS ?, 1 AS ?, 1 AS ?, 1 AS ?,
+            -1.945910149 AS ?, 0 AS ?, 0 AS ?,
+            -1.540445041 AS ?, 0 AS ?, 0 AS ?,
+            -2.63905733 AS ?, 0 AS ?, 0 AS ?,
+            0.1 AS ?, 1000000 AS ?,
+            0.50249 AS ?, 0.70391 AS ?,
+            0.09492 AS ?, 0.11402 AS ?,
+            0.26098 AS ?, 0.14622 AS ?,
+            0.1416 AS ?, 0.03585 AS ?", dr4
+            )

```

Comment:
Here the parameters are still hardcoded, you are replacing the names of the parameters with the csv file, not the values of the parameters